modules/settings/module.php

```diff
@@ -172,17 +154,57 @@ public function on_connect(): void {
 	 */
 	public function save_plan_data( $register_response ) : void {
 		if ( $register_response && ! is_wp_error( $register_response ) ) {
-			$decoded_response = json_decode( $register_response );
-			Data::set_subscription_id( $decoded_response->id );
+			$decoded_response = $register_response;
+			Data::set_subscription_id( $decoded_response->plan->subscription_id );
 			update_option( Settings::PLAN_DATA, $decoded_response );
 			update_option( Settings::IS_VALID_PLAN_DATA, true );
 			$this->set_default_settings();
+			self::set_plan_data_refresh_transient();
 		} else {
 			Logger::error( esc_html( $register_response->get_error_message() ) );
 			update_option( Settings::IS_VALID_PLAN_DATA, false );
 		}
 	}
 
+	/**
+	 * Refresh the plan data after 12 hours
+	 * @return void
+	 */
+	public static function refresh_plan_data() : void {
+
+		if ( ! Connect::is_connected() ) {
+				return;
+		}
+
+	  // Refresh only if refresh transient is expired
+	  if ( self::get_plan_data_refresh_transient() ) {
+		  return;
+	  }
+
+		$plan_data = Settings::get( Settings::PLAN_DATA );
+
+		// Return if plan data does not have public_api_key
+		if ( ! $plan_data->public_api_key ) {
+			Logger::error('Cannot refresh the plan data. No public API key found.');
+			return;
+		}
```

Comment:
then we will have users stuck in this state!
if we don't have an API key you should call register site to get the API key